### ISSUE TITLE
Reports of memory leaks. Requires investigation.

### ISSUE THREAD

**Issue by smashah** [open-wa/wa-automate-nodejs#2050]

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
  console.log(execSync(cmd).toString());
}
//set the version type
process.env.VERS = "minor"

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci $VERS`);

// types only package
exec('npm config set //registry.npmjs.org/:_authToken $NPM_TOKEN')
exec(`cd ./types-only && npm version $VERS && npm run build && npm publish && cd ..`);
exec(`git commit -a -m 'updated types-only package'`);
exec(`git push --force`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [4.20.0](https://github.com/open-wa/wa-automate-nodejs/compare/4.19.3...4.20.0) (2021-08-24)

- build(deps-dev): bump @typescript-eslint/parser from 4.29.2 to 4.29.3 [`#2049`](https://github.com/open-wa/wa-automate-nodejs/pull/2049)
- Release 4.20.0 [`88c1cd1`](https://github.com/open-wa/wa-automate-nodejs/commit/88c1cd1c122f01427df22452d21c9159fb3138fc)
- ✨ feat: `config.maxChats` & `config.maxMessages` #2050 [`a45c4b3`](https://github.com/open-wa/wa-automate-nodejs/commit/a45c4b325b1121cdae7f4a8ed542c080b25a4d4c)
- 🔧 CLI: help docs were showing string for number flags. fixed [`0cdbfbb`](https://github.com/open-wa/wa-automate-nodejs/commit/0cdbfbba01f6f2e5721434b169ed51b8306a26a3)
- updated types-only package [`3415353`](https://github.com/open-wa/wa-automate-nodejs/commit/34153530160b4df406952c534cf5b5ccaf92088e)
- 🗃 memory optimisations #2050 [`c8d004b`](https://github.com/open-wa/wa-automate-nodejs/commit/c8d004baa5d9406265e839f44e14f9dd0d73ef2e)

**Comment by smashah** [comment]
Going forward this library will, by default, only allow 1000 chats and 1000 messages to be "cached" in the session at any time. In order to change these numbers set `config.maxChats` and `config.maxMessages` respectively. Set to `0` to disable this feature.

CLI:

```
//set to only 50 chats and only 50 messages
> npx @open-wa/wa-automate --max-chats 50 --max-messages 50

//Disable session memory optimisation
> npx @open-wa/wa-automate --max-chats 0 --max-messages 0
```
